src/world/Actor/PlayerQuest.cpp

```diff
@@ -1083,29 +1083,22 @@ bool Sapphire::Entity::Player::giveQuestRewards( uint32_t questId, uint32_t opti
   {
     for( uint32_t i = 0; i < rewardItemCount; i++ )
     {
-      auto itemId = questInfo->itemReward0.at( i );
+      auto itemId = questInfo->itemReward.at( i );
       if( itemId > 0 )
       {
-        addItem( itemId, questInfo->itemCountReward0.at( i ), false, false, true, true );
+        addItem( itemId, questInfo->itemCountReward.at( i ), false, false, true, true );
       }
     }
   }
 
-  auto isHq = false;
-  if( optionalChoice > 1000000 )
-  {
-    optionalChoice -= 1000000;
-    isHq = true;
```

Comment:
I think you want to keep this, unless it is actually changed in EW.

Reply:
I tried tracking down where that originated. And looked at the blame for it. You're definitely right about keeping it.